TaskfMRIAnalysis/TaskfMRIAnalysis.sh

```diff
@@ -39,56 +39,92 @@
-  exit 1
-fi
+#arguments to opts_Add*: switch, variable to set, name for inside of <> in help text, description, [default value if AddOptional], [compatibility flag, ...]
+#help info for option gets printed like "--foo=<$3> - $4"
+opts_AddMandatory '--study-folder' 'Path' '/path/to/study/folder' "directory containing imaging data for all subjects"
+opts_AddMandatory '--subject' 'Subject' 'SubjectID' ""
+opts_AddMandatory '--lvl1tasks' 'LevelOnefMRINames' 'tfMRI_TaskName_RL@tfMRI_TaskName_LR' "List of subdirectories containing each time series image for the TaskName task. Multiple scan subdirectories should be provided as a single string separated by '@' character." #Assumes these subdirectories are located in the SubjectID/MNINonLinear/Results directory. Also assumes that timeseries image filename begins with this string.
+opts_AddOptional '--lvl1fsfs' 'LevelOnefsfNames' 'DesignName_TaskName_RL@DesignName_TaskName_LR' "List of prefixes in the design.fsf filenames for each time series image for the TaskName task. Should contain same number of design files as time series images in --lvl1tasks option. (N-th design will be used for N-th time series image.) Separate multiple design files by '@' character. If no value is passed to --lvl1fsfs, the value will be set to the same list passed to --lvl1tasks."
+opts_AddOptional '--lvl2task' 'LevelTwofMRIName' 'tfMRI_TaskName' "Name of Level2 subdirectory in which all Level2 feat directories are written for TaskName. Default is 'NONE', which means that no Level2 analysis will run." 'NONE'
+opts_AddOptional '--lvl2fsf' 'LevelTwofsfName' 'DesignName_TaskName' "Prefix of design.fsf filename for the Level2 analysis for TaskName. If no value is passed to --lvl2fsf, the value will be set to the same list passed to --lvl2task."
+opts_AddOptional '--lowresmesh' 'LowResMesh' 'integer' "Default=32, which is appropriate for HCP minimal preprocessing pipeline outputs" '32'
+opts_AddOptional '--grayordinatesres' 'GrayordinatesResolution' 'integer' "Default=2mm, which is appropriate for HCP minimal preprocessing pipeline outputs" '2'
+opts_AddOptional '--origsmoothingFWHM' 'OriginalSmoothingFWHM' 'integer' "Default=2mm, which is appropriate for HCP minimal preprocessing pipeline outputs" '2'
+opts_AddOptional '--finalsmoothingFWHM' 'FinalSmoothingFWHM' 'integer' "Default=2mm, which is no additional smoothing above HCP minimal preprocessing pipelines outputs. Total desired smoothing (finalsmoothingFWHM) is reached by applying additional smoothing onto origsmoothingFWHM." '2'
+opts_AddOptional '--confound' 'Confound' 'filename' "Confound matrix text filename (e.g., output of fsl_motion_outliers). Assumes file is located in SubjectID/MNINonLinear/Results/tfMRI_TaskName_Scan. Default='NONE'" 'NONE'
+opts_AddOptional '--highpassfilter' 'TemporalFilter' 'integer' "Value for nonlinear highpass filter (in seconds). Default=200. Use 2000 for linear detrend." '200'
```

Comment:
I wouldn't call 2000 a linear detrend.  Just leave that bit out.

Reply:
Also, would be good to make clearer that this represents _additional_ temporal smoothing to be applied during task processing, esp. since (as I understand it), there is going to be argument added for specifying the hp filter (or filter string) that was already used previously (e.g., ICAFIX).

Reply:
Changing to: 
"Apply *additional* highpass filter (in seconds) to time series and task design. This is above and beyond temporal filter applied during preprocessing. Default=200."